### ISSUE TITLE
How to set the order and their theta for multiply batch variates?

### ISSUE THREAD

**Issue by QqQss** [immunogenomics/harmony#24]
Hi, guys, thanks for your great tool!

I have a very large data comprised several levels of batches which are 3 platform, 10+ datasets and 100+ patients. 
I want to use harmony to reduce these unwant variances, but I don't know how to set the order and their weights (i.e. 'theta') for my data. Do you have any suggestion?

BTW, does the order of covariates would affect harmony's result?

**Comment by veghp** [comment]
I'm also interested in this. Does the order of batch variables matter? How should it be set? I looked into the documentation and the R code, but it's not clear from that what's happening. I have two nested factors; donors and treatments.

**Comment by ilyakorsunsky** [comment]
@QqQss, @veghp, great question! The order of variables does not matter. In my experience, I've used theta values between 0 and 4 for a single variable. For multiple variables, thetas are additive. For instance, let's say I set theta=c(2,3) for donor and platform. Groups of cell with the same platform but from different donors will be merged with an effective theta=2. Whereas cells with different platform and different donors will be merged with an effective theta=5. 

So far, I don't have a systematic way of setting thetas. I usually try 2 or 3 different ones and find that there is some value of theta where the solutions start to converge. For example, merging different 10X runs I use theta=0, 1, and 2, and find theta=1 and theta=2 give very similar answers.
